includes/admin/class-llms-sendwp.php

```diff
@@ -0,0 +1,399 @@
+					'<a href="' . admin_url( '/tools.php?page=sendwp' ) . '">',
+					'</a>'
+				);
+			}
+
+			return '<p>' . implode( ' ', $ret ) . '</p>';
+
+		}
+
+		return '<button class="button button-primary" id="llms-sendwp-connect"><i class="fa fa-paper-plane-o" aria-hidden="true"></i> Connect SendWP</button>';
+
+	}
+
+	/**
+	 * Determine if inline scripts and styles should be output.
+	 *
+	 * @since [version]
+	 *
+	 * @return bool
+	 */
+	protected function should_output_inline() {
+
+		// Short circuit if missing unauthorized.
+		if ( ! current_user_can( 'install_plugins' ) ) {
+			return false;
+		}
+
+		$screen = get_current_screen();
+		return ( 'lifterlms_page_llms-settings' === $screen->id && 'engagements' === llms_filter_input( INPUT_GET, 'tab', FILTER_SANITIZE_STRING ) );
+
+	}
+
+	/**
+	 * Output some quick and dirty inline CSS.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function output_css() {
+
+		if ( ! $this->should_output_inline() ) {
+			return;
+		}
+
+		?>
+		<style type="text/css">
+			#llms-sendwp-connect {
+				font-size: 16px;
+				height: auto;
+				margin: 0 0 6px;
+				padding: 8px 14px;
+			}
+			#llms-sendwp-connect .fa {
+				margin-right: 4px;
+			}
+		</style>
+		<?php
+	}
+
+
+	/**
+	 * Output some quick and dirty inline JS.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function output_js() {
+
+		if ( ! $this->should_output_inline() ) {
+			return;
+		}
+
+		?>
+		<script>
+			var btn = document.getElementById( 'llms-sendwp-connect' )
+			btn.addEventListener( 'click', function( e ) {
+				e.preventDefault();
+				llms_sendwp_remote_install();
+			} );
+
+			/**
+			 * Perform AJAX request to install SendWP plugin.
+			 *
+			 * @since [version]
+			 *
+			 * @return void
+			 */
+			function llms_sendwp_remote_install() {
+				var data = {
+					'action': 'llms_sendwp_remote_install',
+				};
+				jQuery.post( ajaxurl, data, function( res ) {
+					llms_sendwp_register_client( res.register_url, res.client_name, res.client_secret, res.client_redirect, res.partner_id );
+				} ).fail( function( jqxhr ) {
+					$( btn ).parent().find( '.llms-error' ).remove();
+					if ( jqxhr.responseJSON && jqxhr.responseJSON.message ) {
+						$( '<p class="llms-error">' + LLMS.l10n.replace( 'Error: %s', { '%s': jqxhr.responseJSON.message } ) + '</p>' ).insertAfter( $( btn ) );
+						console.log( jqxhr );
+					}
+				} );
+			}
+
+			/**
+			 * Register client with SendWP.
+			 *
+			 * @since [version]
+			 *
+			 * @param {string} register_url Registration URL.
+			 * @param {string} client_name Client name.
+			 * @param {string} client_secret Client secret.
+			 * @param {string} client_redirect Client redirect URL.
+			 * @param {int} partner_id SendWP partner ID.
+			 * @return {void}
+			 */
+			function llms_sendwp_register_client( register_url, client_name, client_secret, client_redirect, partner_id ) {
```

Comment:
never considered this as the easiest way to post+goto :D great!

Reply:
SendWP devs from SaturdayDrive came up with that. Can't take credit :-(